aws/resource_aws_ecs_service.go

```diff
@@ -489,7 +508,11 @@ func resourceAwsEcsServiceRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("task_definition", taskDefinition)
 	}
 
-	d.Set("desired_count", service.DesiredCount)
```

Comment:
In my testing, it's the number of container instances, which can change periodically.

Reply:
Adding the following [`DiffSuppressFunc`](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#diffsuppressfunc) to the `desired_count`, `deployment_minimum_percent`, and `deployment_maximum_percent` attributes, we can continue to always call `d.Set("desired_count", service.DesiredCount)` (along with the others) and remove the conditional in the read function. This will remove the breaking change for outputs that depend on the `desired_count` attribute always existing.

```go
DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
	if d.Get("scheduling_strategy").(string) == ecs.SchedulingStrategyDaemon {
		return true
	}
	return false
},
```